saleor/dashboard/forms.py

```diff
@@ -33,24 +43,58 @@ def valid_value(self, value):
         forms.Field.validate(self, value)
         return True
 
-    def set_initial(self, obj):
+    def set_initial(self, obj, obj_id=None, label=None):
         """Set initially selected objects on field's widget."""
-        selected = {'id': obj.pk, 'text': str(obj)}
+        selected = {
+            'id': obj_id if obj else obj.pk,
+            'text': label if label else str(obj)}
         self.widget.attrs['data-initial'] = json.dumps(selected)
 
 
+class AjaxSelect2CombinedChoiceField(AjaxSelect2ChoiceField):
+    """An AJAX-based choice field using Select2 for multiple querysets.
+
+    <obj.pk>_<obj.__class__.__name__> represents value passed to a field.
+    fetch_data_url - specifies url, from which select2 will fetch data
+    initial - initial object
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.querysets = kwargs.pop('querysets')
+        super().__init__(*args, **kwargs)
+
+    def to_python(self, value):
+        if value in self.empty_values:
+            return None
+
+        pk, model_name = value.split('_')
+        queryset = next(
+            (qs for qs in self.querysets if qs.model.__name__ == model_name),
+            None)
+
+        value = queryset.filter(pk=pk).first() if queryset else None
```

Comment:
Can we use `get(pk=pk)` here? Would be a bit more readable if does not break the code.

Reply:
Then we should have `try`/`except` clause. I think that above statement is simpler.

Reply:
`get()` throws an error when a key is not found while `filter.first()` returns `None` which is handled below.